MetaMask/metamask-extension
# Update Icon to use TS Box

## Explanation
Currently the `Icon` uses the JavaScript version of the `Box` component. This PR updates it to use the TypeScript version.
- Updated `Icon` to use the TS version of the `Box` component
- Updated any component types that use a form of IconProps to use the new generic type format
- Updates snapshot tests to use new `mm-box` class and removes unused default flex-direction class from old JS Box version
* Fixes #19989

## Screenshots/Screencaps
There should be no style updates included in this PR so the below screencasts so no visual regressions

### Before

https://github.com/MetaMask/metamask-extension/assets/8112138/9b774a57-440d-469a-a028-cb8114548697

### After

https://github.com/MetaMask/metamask-extension/assets/8112138/90d1c37d-0198-44db-af75-1863159b7b5a

## Manual Testing Steps

- Go to the storybook build of this PR
- Search `Icon` (make sure the component-library version is selected)
- Check documentation and stories for no visual regressions

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
